main_test.go

```diff
@@ -95,3 +95,60 @@ func TestCopyFromCommandLineArgument(t *testing.T) {
 		t.Errorf("Expected %s but got %s", content, copied)
 	}
 }
+
+func TestParseContentDirectText(t *testing.T) {
+	directText := "Direct text input"
+	contentStr, err := parseContent(&directText, []string{})
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+	if contentStr != directText {
+		t.Errorf("Expected contentStr to be '%s', got '%s'", directText, contentStr)
+	}
+}
+
+func TestParseContentStdin(t *testing.T) {
+	stdinInput := "Stdin input\n"
+	r, w, _ := os.Pipe()
+	originalStdin := os.Stdin
+	defer func() {
+		os.Stdin = originalStdin
+		r.Close()
+		w.Close()
+	}()
+	os.Stdin = r
+
+	// Write the input to the pipe
+	w.WriteString(stdinInput)
+	w.Close()
+
+	contentStr, err := parseContent(nil, []string{})
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+	if contentStr != stdinInput {
+		t.Errorf("Expected contentStr to be '%s', got '%s'", stdinInput, contentStr)
+	}
+}
+
+func TestParseContentFromFile(t *testing.T) {
+	contentFromFile := "Content from file"
+	fakeFile := createTempFile(contentFromFile)
```

Comment:
```suggestion
	fakeFile := createTempFile(t, contentFromFile)
```

Please have a look at https://github.com/supitsdu/clipper/pull/17#discussion_r1651958704

Then the defer can be removed

Reply:
Always nice to have feedbacks like these, constructive and straightforward. This reviews seems to add a proper resource cleanup and error handling, making the test suite safer and easier to maintain. Loved it!

Reply:
You are welcome, I'm glad you like my feedbacks